### ISSUE TITLE
[BUG][Java | JaxRS-Jersey] Enum Name instead of value used to render JSON

### ISSUE THREAD

**Issue by rkarallus** [OpenAPITools/openapi-generator#2468]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description
We are generating our Model using the OpenAPI Generator. In case of an enum the generated code behaves incorrect. Instead of rendering the enum values it renders the enum names to the JSON which results in incorrect and unexpected values if using lowercase or numeric values.

##### openapi-generator version
"org.openapitools:openapi-generator-cli:3.3.4"

##### Command line used for generation
```
configurations {
    openAPIGenerator
}

dependencies {
    openAPIGenerator "org.openapitools:openapi-generator-cli:3.3.4"
}

task runOpenAPIGenerator(type: JavaExec) {
    main = 'org.openapitools.codegen.OpenAPIGenerator'
    classpath = configurations.openAPIGenerator
    args 'generate','-i','...','-g','jaxrs-jersey','-o','.','-Dmodels','-DmodelDocs=false','--model-package','...'
}
```
##### Steps to reproduce
```
    ErrorCode:
      type: string
      enum: [1,2,3,4,500]
```
Generating this type will result in the following enum:
```
public enum ErrorCode {
  _1("1"),
  _2("2"),
  _3("3"),
  _4("4"),
  _500("500");
  ...
}
```
If the ErrorCode is used in a response entity the respective JSON will be
```
    {
        "code": "_1"
    }
```
If using numbers the effect is even worse as the resulting data type in the JSON will be wrong
```
ErrorCode:
      type: integer
      enum: [1,2,3,4,500]
```
Generated Class
```
public enum ErrorCode {
  NUMBER_1(1),
  NUMBER_2(2),
  NUMBER_3(3),
  NUMBER_4(4),
  NUMBER_500(500);
```
Resulting JSON
```
    {
        "code": "NUMBER_1"
    }
```
##### Suggest a fix

I think this stackoverflow article points to a couple of possible solutions:
https://stackoverflow.com/questions/12468764/jackson-enum-serializing-and-deserializer

**Comment by karismann** [comment]
your bug seem to be fixed with PR #1833  so please try with the version 4.0.0-beta2

**Comment by rkarallus** [comment]
I agree. the bug is solved in 4.0.0-beta2. Tried it out, works fine.
